application/common/models/Mfa.php

```diff
@@ -55,20 +63,23 @@ public function fields(): array
                 }
                 return null;
             },
-            'data' => function ($model) {
-                $data = [];
-                /** @var Mfa $model */
-                if ($model->verified === 1 && $model->scenario === User::SCENARIO_AUTHENTICATE) {
-                    $data += $model->authInit();
-                }
-                if ($model->type === self::TYPE_BACKUPCODE || $model->type === self::TYPE_MANAGER) {
-                    $data += ['count' => count($model->mfaBackupcodes)];
-                }
-                return $data;
-            }
+            'data',
         ];
     }
 
+    /**
+     * @param string $rpOrigin The Relay Party Origin, used for WebAuthn and ignored for others
+     */
+    public function loadData(string $rpOrigin = '') {
```

Comment:
I'm wondering if this should be `public static function` and also have the curly bracket drop down a line.

Reply:
it cant be static since it uses $this to update data. but you're right the brace should be down a line